File: DeJia/app/src/main/java/com/dejia/anju/api/base/BaseNetWorkCallBackApi.java
package com.dejia.anju.api.base;

import android.content.Context;

import com.dejia.anju.net.NetWork;
import com.dejia.anju.net.ServerData;

import java.util.HashMap;

public class BaseNetWorkCallBackApi {

    private HashMap<String, Object> mHashMap;  //传值容器
    private final String mController;
    private final String mMethodName;
    private Context context;
    private String TAG = "BaseNetWorkCallBackApi";

    public BaseNetWorkCallBackApi(String controller, String methodName,Context mContext) {
        mHashMap = new HashMap<>(0);
        this.mController = controller;
        this.mMethodName = methodName;
        this.context = mContext;
    }

    /**
     * 开始请求
     *
     * @param listener
     */
    public void startCallBack(final BaseCallBackListener<ServerData> listener) {
        NetWork.getInstance().call(mController, mMethodName, mHashMap, context, mData -> listener.onSuccess(mData));
    }

    public void addData(String key, String value) {
        mHashMap.put(key, value);
    }

    public HashMap<String, Object> getHashMap() {
        return mHashMap;
    }

}

File: DeJia/app/src/main/java/com/dejia/anju/api/base/BaseNetWorkCallBackApi.kt
package com.dejia.anju.api.base

import android.content.Context
import com.dejia.anju.net.NetWork.Companion.instance
import com.dejia.anju.net.ServerData
import java.util.*

class BaseNetWorkCallBackApi(controller: String, methodName: String, mContext: Context) {
    val hashMap //传值容器
            : HashMap<String, Any>
    private val mController: String
    private val mMethodName: String
    private val context: Context
    private val TAG = "BaseNetWorkCallBackApi"

    /**
     * 开始请求
     *
     * @param listener
     */
    fun startCallBack(listener: BaseCallBackListener<ServerData?>) {
        instance!!.call(mController, mMethodName, hashMap, context) { mData: ServerData? -> listener.onSuccess(mData) }
    }

    fun addData(key: String, value: String) {
        hashMap[key] = value
    }

    init {
        hashMap = HashMap(0)
        mController = controller
        mMethodName = methodName
        context = mContext
    }
}